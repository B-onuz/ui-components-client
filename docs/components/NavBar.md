# NavBar (Component)

Componente de barra superior para layouts.

## Como utilizar

```js
import NavBar from 'ui-components/lib/components/NavBar'
```

```jsx
<NavBar>Foo Bar</NavBar>
```
