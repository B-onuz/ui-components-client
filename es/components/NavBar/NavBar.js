"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _this = void 0,
    _jsxFileName = "/Users/master/projects/bonuz/ui-components/src/components/NavBar/NavBar.js";

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n  background-color: #f4f4f4;\n  height: 64px;\n  display: flex;\n  align-items: center;\n  padding: 20px;\n  box-sizing: border-box;\n  flex-shrink: 0;\n  @media only screen and (max-width: 600px) {\n    height: auto;\n    width: 100%;\n    flex-direction: column;\n    justify-content: center;\n    padding: 2vh;\n    position: fixed;\n    width: 100%;\n    z-index: 1000;\n    flex-wrap: wrap;\n    & > header {\n      max-width: 85%;\n      & > h1 {\n        max-width: 100%;\n        margin-bottom: 8px;\n        overflow: hidden;\n        white-space: nowrap;\n        text-overflow: ellipsis;\n      }\n    }\n    & > button {\n      position: absolute;\n      right: 8px;\n      top: 33.33%;\n    }\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteralLoose(strings, raw) { if (!raw) { raw = strings.slice(0); } strings.raw = raw; return strings; }

var StyledNavBar = _styledComponents["default"].div(_templateObject());

var NavBar = function NavBar(_ref) {
  var rest = _extends({}, _ref);

  return /*#__PURE__*/_react["default"].createElement(StyledNavBar, _extends({}, rest, {
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 41,
      columnNumber: 10
    }
  }));
};

var _default = NavBar;
exports["default"] = _default;