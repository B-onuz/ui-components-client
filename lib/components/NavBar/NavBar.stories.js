'use strict'

exports.__esModule = true
exports.NavBarExample = exports['default'] = void 0

var _react = _interopRequireDefault(require('react'))

var _NavBar = _interopRequireDefault(require('./NavBar'))

var _this = void 0,
  _jsxFileName = '/home/junior/Fontes/ui-components-client/src/components/NavBar/NavBar.stories.js'

function _interopRequireDefault(obj) {
  return obj && obj.__esModule ? obj : { default: obj }
}

var _default = {
  title: 'NavBar',
  component: _NavBar['default'],
}
exports['default'] = _default

var NavBarExample = function NavBarExample() {
  return /*#__PURE__*/ _react['default'].createElement(
    _NavBar['default'],
    {
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 9,
        columnNumber: 36,
      },
    },
    'lorem'
  )
}

exports.NavBarExample = NavBarExample
