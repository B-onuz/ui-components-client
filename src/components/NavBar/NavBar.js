import React from 'react'
import styled from 'styled-components'

const StyledNavBar = styled.div`
  background-color: #FFFFFF;
  display: flex;
  align-items: center;
  padding: 20px 40px 10px 40px;
  box-sizing: border-box;
  flex-shrink: 0;
  @media only screen and (max-width: 600px) {
    height: auto;
    width: 100%;
    flex-direction: column;
    justify-content: center;
    padding: 2vh;
    position: fixed;
    width: 100%;
    z-index: 1000;
    flex-wrap: wrap;
    & > header {
      max-width: 85%;
      & > h1 {
        max-width: 100%;
        margin-bottom: 8px;
        overflow: hidden;
        white-space: nowrap;
        text-overflow: ellipsis;
      }
    }
    & > button {
      position: absolute;
      right: 8px;
    }
  }
`

const NavBar = ({ ...rest }) => {
  return <StyledNavBar {...rest} />
}

export default NavBar