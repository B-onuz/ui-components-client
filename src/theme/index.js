const colors = {
  red: '#db2828',
  lightRed: '#ef4d5a',
  mediumRed: '#FF4F5D',
  darkRed: '#9c2543',
  fullDarkRed: '#550e33',
  orange: '#f2711c',
  yellow: '#fbbd08',
  olive: '#b5cc18',
  green: '#21BA45',
  teal: '#00b5ad',
  blue: '#2185d0',
  facebookBlue: '#3B5998',
  blueLink: '#4285F4',
  violet: '#6435c9',
  purple: '#a333c8',
  pink: '#e03997',
  brown: '#a5673f',
  grey: '#767676',
  textGrey: '#8D8D8D',
  lightGrey: '#6D6D6D',
  mediumGrey: '#838383',
  darkGrey: '#9A9A9A',
  boldGray: '#55555A',
  black: '#1b1c1d',
  white: '#fdfdfd',
  warning: '#7A4D05',
  success: '#1E561F',
  info: '#0E566C',
  danger: '#973937',
  cyan: '#1CB5AD',
  disabledGrey: '#ffffff1f',
  disabledGreyText: '#ffffff4d',
}

colors.primary = colors.lightRed
colors.secondary = colors.fullDarkRed
colors.default = '#e0e1e2'

const breakpoints = ['40em', '52em', '64em', '80em']

breakpoints.sm = breakpoints[0]
breakpoints.md = breakpoints[1]
breakpoints.lg = breakpoints[2]
breakpoints.xl = breakpoints[3]

const fontSizes = [12, 14, 16, 20, 24, 32]

export default {
  fontFamily: `'Lato', sans-serif`,
  colors,
  fontSizes,
  breakpoints,
}
