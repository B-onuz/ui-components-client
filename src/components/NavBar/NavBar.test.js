import React from 'react'
import NavBar from './NavBar.js'
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'
import theme from '../../theme'
import 'jest-styled-components'

test('NavBar changes the class when hovered', () => {
  const component = renderer.create(
    <ThemeProvider theme={theme}>
      <NavBar />
    </ThemeProvider>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
