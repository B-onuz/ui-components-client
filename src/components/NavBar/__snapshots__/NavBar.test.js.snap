// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar changes the class when hovered 1`] = `
.c0 {
  background-color: #f4f4f4;
  height: 64px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 20px;
  box-sizing: border-box;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

@media only screen and (max-width:500px) {
  .c0 {
    -webkit-flex-direction: row-reverse;
    -ms-flex-direction: row-reverse;
    flex-direction: row-reverse;
    padding: 2vh;
  }
}

<div
  className="c0"
/>
`;
